ipfshttpclient/client/dag.py

```diff
@@ -0,0 +1,149 @@
+import typing as ty
+
+from . import base
+
+from .. import multipart
+
+
+class Section(base.SectionBase):
+	@base.returns_single_item(base.ResponseBase)
+	def get(self, cid: str, **kwargs: base.CommonArgs):
+		"""Get and serialize the DAG node named by CID.
+
+		.. code-block:: python
+
+			>>> client.dag.get('QmTkzDwWqPbnAh5YiV5VwcTLnGdwSNsNTn2aDxdXBFca7D')
+			{'Data': '\x08\x01',
+			 'Links': [
+				{'Hash': 'Qmd2xkBfEwEs9oMTk77A6jrsgurpF3ugXSg7dtPNFkcNMV',
+				 'Name': 'Makefile',          'Size': 174},
+				{'Hash': 'QmeKozNssnkJ4NcyRidYgDY2jfRZqVEoRGfipkgath71bX',
+				 'Name': 'example',           'Size': 1474},
+				{'Hash': 'QmZAL3oHMQYqsV61tGvoAVtQLs1WzRe1zkkamv9qxqnDuK',
+				 'Name': 'home',              'Size': 3947},
+				{'Hash': 'QmZNPyKVriMsZwJSNXeQtVQSNU4v4KEKGUQaMT61LPahso',
+				 'Name': 'lib',               'Size': 268261},
+				{'Hash': 'QmSY8RfVntt3VdxWppv9w5hWgNrE31uctgTiYwKir8eXJY',
+				 'Name': 'published-version', 'Size': 55}
+			]}
+
+		Parameters
+		----------
+		cid
+			Key of the object to retrieve, in CID format
+
+		Returns
+		-------
+			dict
+				Cid with the address of the dag object
+		"""
+		args = (str(cid),)
+		return self._client.request('/dag/get', args, decoder='json', **kwargs)
+
+	@base.returns_single_item(base.ResponseBase)
+	def put(self, data: ty.IO, **kwargs: base.CommonArgs):
+		"""Stores input as a DAG object and returns its key.
+
+		.. code-block:: python
+
+			>>> client.dag.put(io.BytesIO(b'''
+			...       {
+			...           "Data": "another",
+			...           "Links": [ {
+			...               "Name": "some link",
+			...               "Hash": "QmXg9Pp2ytZ14xgmQjYEiHjVjMFXzCV … R39V",
+			...               "Size": 8
+			...           } ]
+			...       }'''))
+			{'Cid': {
+					'/': 'bafyreifgjgbmtykld2e3yncey3naek5xad3h4m2pxmo3of376qxh54qk34'
+				}
+			}
+
+		Parameters
+		----------
+		data
+			IO stream object with data that should be put
+
+		Returns
+		-------
+			dict
+				Cid with the address of the dag object
+		"""
+		body, headers = multipart.stream_files(data, chunk_size=self.chunk_size)
+		return self._client.request('/dag/put', decoder='json', data=body,
+		                            headers=headers, **kwargs)
+
+	@base.returns_single_item(base.ResponseBase)
+	def resolve(self, cid: str, **kwargs: base.CommonArgs):
+		"""Resolves a DAG node from its Cid
+
+		.. code-block:: python
+
+			>>> client.dag.resolve('QmTkzDwWqPbnAh5YiV5VwcTLnGdwSNsNTn2aDxdXBFca7D')
+			{'Cid': {
+					'/': 'QmTkzDwWqPbnAh5YiV5VwcTLnGdwSNsNTn2aDxdXBFca7D'
+				}
+			}
+
+		Parameters
+		----------
+		cid
+			Key of the object to resolve, in CID format
+
+		Returns
+		-------
+			dict
+				Cid with the address of the dag object
+		"""
+		args = (str(cid),)
+		return self._client.request('/dag/resolve', args, decoder='json', **kwargs)
+
+	@base.returns_single_item(base.ResponseBase)
+	def imprt(self, data: ty.IO, **kwargs: base.CommonArgs):
+		"""Imports a .car file with a DAG to IPFS
```

Comment:
This should include a comment about why the name is spelled incorrectly (it wasn't immediately obvious to me and I'm sure there are lots of less experienced Python programmers that wouldn't get it all and just be annoyed by what they think is a typo).